### DISCUSSION TITLE
[discussion] Chunk uncles

### DISCUSSION THREAD

**Discussion by SkidanovAlex** [near/nearcore#3945] [Ideas]
I propose the following addition to the protocol:

A chunk at height `H+1` can include a hash and a signature of a chunk at height `H` if the block at height `H` is missing the chunk. If such a chunk and a signature are included, the number of chunks produced by the chunk producer at `H` is increased by 0.8.

This effectively doubles the time that the chunk producers has to have their chunk distributed to not be fully penalized for missing it. And also it doubles the number of participants to whom the message needs to be lost for the full penalty.

**Comment by bowenwang1996** [comment]
When is the chunk at height `H` applied if we do this?

**Comment by Kouprin** [comment]
> When is the chunk at height `H` applied if we do this?

Very good question. I see two possibilities of what @SkidanovAlex meant:
1. Chunk of height `H+1` has all txs from chunk of height `H`, so there is no need to apply the chunk of height `H`.
2. Chunk of height `H` should be applied right before applying chunk of height `H+1`.

In both understanding there is a gap for block producers to decide what to do with current chunk:
1. Include the chunk, or
2. Not include the chunk intentionally and include it at the next height.

Block hash can be affected for unfair purposes, however we never rely that it must be truly random. Sharding cases should work well with possible latency to two blocks (one for skipped txs and one for skipped receipt). Not sure what's happening on epoch boundaries, it's important to look into them too.

**Comment by bowenwang1996** [comment]
I discussed with @SkidanovAlex offline. What alex wants is to not apply that chunk at all and only include chunk hash and signature. The concern I have is that validators are disincentivized to do this because by unless the missed one is produced by themselves, not doing this means that some other validator may get less reward or kicked out, which implies that fewer tokens are minted and is thus more preferable.

**Comment by bowenwang1996** [comment]
@SkidanovAlex is 0.8 calculated based on the current threshold of 0.9? We may want to lower the threshold for chunk producers when there are multiple shards
